pkg/apis/extensions/types.go

```diff
@@ -268,6 +268,11 @@ const (
 
 	// Replace the old RCs by new one using rolling update i.e gradually scale down the old RCs and scale up the new one.
 	RollingUpdateDeploymentStrategyType DeploymentStrategyType = "RollingUpdate"
+
+	// The Deployment controller is not going to orchestrate ReplicaSets for
+	// Deployments that use this strategy. Useful for delegating orchestration of
+	// certain Deployments to external controllers.
+	NoopDeploymentStrategyType DeploymentStrategyType = "Noop"
```

Comment:
@smarterclayton i'm fine with this concept, but also think that 'no-op' is not the right wording (if the deployment update is handled by external controller, it is not really a "no-op"). I can think of names like: "external", "delegated", "custom", "detached"...